Fixes #533 - changing to retain the keyword searched instead final url

marchiore opened the pull request:
This PR is a continuation of an old PR (#544) and it fixes #533.

@oliviabrown9 can you check if everything is ok now?  if you find something wrong i'll be glad in adjust it.

oliviabrown9 commented on Blockzilla.xcodeproj/project.pbxproj:
```diff
@@ -53,7 +54,7 @@
 		74F94FF21FD9CA070047E629 /* Intro.strings in Resources */ = {isa = PBXBuildFile; fileRef = 74F94FF41FD9CA070047E629 /* Intro.strings */; };
 		A57245250F88CCA4993E55B2 /* CoreText.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = F4C4943B406FCA9B74B4E186 /* CoreText.framework */; };
-		A89766DA1F57DCA9008183C5 /* BuildFile in Resources */ = {isa = PBXBuildFile; };
+		A89766DA1F57DCA9008183C5 /* (null) in Resources */ = {isa = PBXBuildFile; };
```
I'm a little bit confused by why/where this change is coming from.

oliviabrown9 commented on Blockzilla/URLBar.swift:
```diff
@@ -470,6 +470,8 @@ class URLBar: UIView {
     func fillUrlBar(text: String) {
         urlText.text = text
     }
+    
```
Nit: remove unnecessary whitespace changes.

oliviabrown9 commented on Blockzilla/URLBar.swift:
```diff
@@ -662,17 +664,21 @@ class URLBar: UIView {
+            }
+            
             truncatedURL = components?.host
-            urlText.text = isEditing ? fullUrl : truncatedURL
```
This reverts one of the 7.0 PRs that displays a truncated version of the URL except when editing. We now need to have three states (search query if a search, full URL when editing, truncated url otherwise). All URLs should still display and act exactly the same as before, only search queries changing.

oliviabrown9 commented on Blockzilla/SearchHistoryUtils.swift:
```diff
@@ -0,0 +1,106 @@
+        return ""
+    }
+    
+    // go back
```
I don't think this comment adds anything.

oliviabrown9 commented on Blockzilla/SearchHistoryUtils.swift:
```diff
@@ -0,0 +1,106 @@
+        }
+    }
+    
+    // go foward
```
I don't think this comment adds anything.

oliviabrown9 commented on Blockzilla/SearchHistoryUtils.swift:
```diff
@@ -0,0 +1,106 @@
+    }
+    
+    // go back
+    static func goFoward() {
```
Nit: should be "goForward"

oliviabrown9 commented on Blockzilla/SearchHistoryUtils.swift:
```diff
@@ -0,0 +1,106 @@
+            }
+        }
+        
+        return ""
```
Should return nil since it's optional.

oliviabrown9 commented on Blockzilla/URLBar.swift:
```diff
@@ -662,17 +664,21 @@ class URLBar: UIView {
             components?.password = nil
-            fullUrl = components?.url?.absoluteString
+            
+            if let searchedText = SearchHistoryUtils.pullSearchFromStack(), searchedText != "" {
```
After you change the function to return nil, you should be able to remove the check for != ""

oliviabrown9 commented on Blockzilla/SearchHistoryUtils.swift:
```diff
@@ -0,0 +1,106 @@
+    // go back
+    static func goFoward() {
+        isNavigating = true
+        var currentStack = [textSearched]()
```
Can you clarify that this will effectively "erase" the stack on each new session? I might be more comfortable having an explicit clear of the stack when the user presses erase -> I think it goes along with the idea that we clear everything we have on the user when they press erase.

marchiore replied:
I'm cleaning the stack on `BrowserViewController.swift`in line `439`.

oliviabrown9 commented on Blockzilla.xcodeproj/project.pbxproj:
```diff
@@ -1428,7 +1431,7 @@
 			files = (
 				742C99D41F3A3AD200717D69 /* Assets.xcassets in Resources */,
 				16938C0121010C2D00DCD489 /* InfoPlist.strings in Resources */,
-				A89766DA1F57DCA9008183C5 /* BuildFile in Resources */,
```
I'm a little bit confused by why/where this change is coming from.

marchiore replied:
Fixed

marchiore commented:
@oliviabrown9, Thank You for your notes! I've updated my PR and comment your notes above. 

If something wrong, let me know! 

:)

oliviabrown9 commented on Blockzilla/URLBar.swift:
```diff
@@ -664,15 +664,22 @@ class URLBar: UIView {
+            } else {
+                displayText = truncatedURL
+            }
+            urlText.text = isEditing ? fullUrl : displayText
```
Looks like this still isn't working as expected. Rather than checking for http/https (which isn't always a given), you might want to check if it can be a URL. You can verify that your solution works by comparing it to the urls currently displayed when you run v7.0-dev. 

Example: twitter.com (when using autocomplete)
Expected:
<img width="373" alt="image" src="https://user-images.githubusercontent.com/20130899/44108422-353d03c4-9faf-11e8-9b85-ca3d673d5968.png">

Result:
<img width="370" alt="image" src="https://user-images.githubusercontent.com/20130899/44108453-4ff77fbe-9faf-11e8-8329-6a9b9b51e4a1.png">

marchiore commented:
@oliviabrown9, i've changed the validation to check if the string it's a URL, as you suggested, and i've updated my PR, can you check this out?

Thanks again for helping me out to solve this issue!

oliviabrown9 commented:
@marchiore Can you merge in the latest v7.0-dev and ping me when ready for review again?

marchiore commented:
@oliviabrown9 merged!

I've noticed a problem on Autocomplete, where it sometimes put a '/' at the begging of type. Is there someting the team is working on?

P.S.: i've checked on a clean branch and the problem persists.

oliviabrown9 commented:
@marchiore I haven't heard about or seen this issue before. Would you mind reproducing and uploading a photo/gif for us to investigate separately from this PR since it sounds unrelated.
